web/controllers/password_controller.ex

```diff
@@ -0,0 +1,37 @@
+defmodule CodeCorps.PasswordController do
+  use CodeCorps.Web, :controller
+
+  alias CodeCorps.{Emails, Mailer, User, AuthToken}
+  alias Ecto.Changeset
+
+  @doc"""
+  forgot_password should take an email and generate an AuthToken model and send an email
+  """
+  def forgot_password(conn, %{"email" => email}) do
+    with %User{} = user <- Repo.get_by(User, email: email),
+        %Changeset{valid?: true} <- AuthToken.changeset(%AuthToken{}, user) do
+
+        try_create_forgot_password(user)
+        |> send_forgot_password_email
+
+        conn
+        |> put_status(200)
+        |> render("show.json", email: user.email)
+
+    else
+      nil ->
```

Comment:
`with` would actually still be helpful here

```Elixir
with %User{} = user <- Repo.get_by(User, email: email),
     {:ok, %AuthToken{} = token} <- AuthToken.create_changeset(%AuthToken{}, user) |> Repo.insert()
do
  Emails.ForgotPasswordEmail.create(user, token) |> Mailer.deliver_now()
else
  nil -> nil
end

conn |> put_status(:ok) |> render("show.json", email: email)
```

So you render out the same email the user typed in. The with clause is helpful so that you are able to create and send the email if the address the user provided is valid. The response happens anyway, outside the with clause.